### ISSUE TITLE
3 screens + Steam VR = autoswitch to 2 screens

### ISSUE THREAD

**Issue by ganzoni** [terrymacdonald/DisplayMagician#289]
I have an HTC vive pro 2 VR headset with 3 screens and RTX 4090 graphics card.

The VR headset works with SteamVR.

To make SteamVR work, you can have a maximum of 2 screens + the VR headset (impossible with 3 screens + VR = error too much screens).

It would be useful to have an option that detects when the HTC Vive pro 2 HUB is switched on and before it starts SteamVR the system automatically switches to 2 screens.

Is there anything you can do?

**Comment by terrymacdonald** [comment]
Yeah I doubt there is anything I can do. If Steam VR or the Vivo software
acts like a windows display, then you may be able to set it up and save
that layout as a Display Profile. No idea if that will work as I don't have
VR yet.

Other than that I'm not sure if there would be anything software related I
could do. It depends on the VR libraries and how they interact with windows
display. It would be a lot of work to do if it's even possible.

I'm happy to keep this issue open, but it works be a future project to look
at I'm afraid.

Thanks
Terry

**Comment by ganzoni** [comment]
For launching the VR in need to sitch on a HUB, once he is up, he launch SteamVR and Vive software.

At this moment if i have 3 screens on, i have the "Too many screens" error.
Also it's not possible to create a profile.

**Comment by terrymacdonald** [comment]
I don't have any VR to test with I'm afraid. If anyone out there wants to donate their old VR stuff that uses Vive or SteamVR software then I might be able to do some testing, but for now without anything to test I don't have the first idea where to start.

I'm not sure if VR stuff acts like a plain windows display or not. If they do then it should work already, but if they act like something else, then I'll need to figure out what is different about the VR screens and then add support for them into the WinLibrary video card library.

Actually IIRC this was logged before we made the last set of changes, so maybe try the latest DisplayMagician again today and see if that works now? Maybe the other things I fixed will make this work too?

If not please reply back and we'll have to leave this issue for the future.

Thanks
Terry

**Comment by ganzoni** [comment]
You can install "SteamVR"

My idea is that as soon as "vrmonitor.exe" is launched display magician switch to 2 monitors.

May be you can track "vrmonitor.exe"?
<img width="664" alt="image" src="https://github.com/terrymacdonald/DisplayMagician/assets/31613494/fdfefa1f-6c09-4f65-8653-8ed15eef66ee">

**Comment by terrymacdonald** [comment]
It's way way way tricker than simply switching to 2 screens. Firstly how do
we switch on the screens? Which API do I talk to in order to make that
happen? How does VR interact with the existing windows display APIs like
the CCD API or the GDI API?

I also have to think about how all the other VR ecosystems will work with
DisplayMagician as well. Not just SteamVR, but Meta, HTC, etc. I need to
research of there is a standard way that VR systems interface with windows,
and then I need to build a video library that will interface
DisplayMagician with the VR drivers, one Video library for each ecosystem.

It could easily be 6 months of solid work. This is not a decision that I
will take lightly.

I'll have a little think about this over the next couple of weeks to see
how I feel about this, and whether I want to take on that much extra work.
I am still trying to progress DisplayMagician v3.0, and that is enough work
already.

I only have so much free time to give to the
community. I need to keep at least some of it for me to use actually
simracing 😁!

Thanks
Terry
